src/tiledb/cloud/geospatial/ingestion.py

```diff
@@ -0,0 +1,1287 @@
+    :param nodata: float, NODATA value for destination raster
+    :param resampling: string, resampling method,
+        one of None, bilinear, cubic, nearest and average
+    :param append: bool, whether to append to the array
+    :param verbose: verbose logging, defaults to False
+    :param config: dict, configuration to pass on tiledb.VFS
+    :param compressor: dict, serialized compression filter
+    :param id: str, ID for logging, defaults to None
+    :param trace, bool, enable trace logging
+    :param log_uri: log array URI
+    """
+
+    import rasterio
+    import rasterio.merge
+
+    import tiledb
+    from tiledb.cloud.utilities import Profiler
+    from tiledb.cloud.utilities import get_logger_wrapper
+    from tiledb.cloud.utilities import max_memory_usage
+
+    logger = get_logger_wrapper(verbose)
+
+    with tiledb.scope_ctx(config):
+        with Profiler(array_uri=log_uri, id=id, trace=trace):
+            vfs = tiledb.VFS()
+
+            if not append:
+                import rasterio
+                from rasterio.transform import from_origin
+
+                from tiledb import filter
+
+                if (
+                    extents is None
+                    or res is None
+                    or crs is None
+                    or band_count is None
+                    or dtype is None
+                ):
+                    raise ValueError(
+                        """CRS, band count, extents and resolution are
+                        required to create empty raster array"""
+                    )
+
+                # create empty array,
+                dst_profile = {
+                    "count": band_count,
+                    "dtype": dtype,
+                    "driver": "TileDB",
+                    "crs": crs,
+                    "transform": from_origin(extents.minx, extents.maxy, *res),
+                    "width": math.floor((extents.maxx - extents.minx) / res[0]),
+                    "height": math.floor((extents.maxy - extents.miny) / res[1]),
+                    "blockysize": tile_size,
+                    "blockxsize": tile_size,
+                }
+
+                logger.debug("Destination raster profile %r", dst_profile)
+
+                if nodata is not None:
+                    dst_profile.update(
+                        {
+                            "nodata": nodata,
+                        }
+                    )
+
+                if compressor:
+                    compressor_args = dict(compressor)
+                    compressor_name = compressor_args.pop("_name")
+                    if compressor_name:
+                        compressor_args = {
+                            k: None if not v else v for k, v in compressor_args.items()
+                        }
+                        compression_filter = vars(filter).get(compressor_name)(
+                            **compressor_args
+                        )
```

Comment:
A few things here:

1. You can directly `**` dict literals.
2. Instead of `vars().get()`, use `getattr`
3. `first or second` is exactly equivalent to `second if not first else first`.

```python
compression_filter = getattr(filter, compressor_name)(
    **{k: v or None for k, v in compressor_args.items()}
)
```

Reply:
👍